src/control/server/instance.go

```diff
@@ -245,6 +247,29 @@ func (srv *IOServerInstance) Stop(signal os.Signal) error {
 	return nil
 }
 
+// RemoveSocket removes the socket file used for dRPC communication with
+// harness and updates relevant ready states.
+func (srv *IOServerInstance) RemoveSocket() error {
+	fMsg := fmt.Sprintf("removing instance %d socket file", srv.Index())
+
+	dc, err := srv.getDrpcClient()
+	if err != nil {
+		return errors.Wrap(err, fMsg)
+	}
+	srvSock := dc.GetSocketPath()
+
+	if err := checkDrpcClientSocketPath(srvSock); err != nil {
+		return errors.Wrap(err, fMsg)
+	}
+	os.Remove(srvSock)
+
+	// reset state
+	srv.ready.SetFalse()
+	srv.drpcReady = make(chan *srvpb.NotifyReadyReq)
```

Comment:
I don't think we actually need to remake the channel. It looks like we just wait on a message from the channel, and we don't close it. So I think it would still be okay at this point.

Reply:
good catch